File: c/matriz_geral.c
#include <stdio.h>
#include <math.h>

int main() {

    int N, indLinha, indColuna;
    double somaPositivos;

    printf("Qual a ordem da matriz? ");
    scanf("%d", &N);

    double mat[N][N];

    for (int i = 0; i < N; i++) {
        for (int j = 0; j < N; j++) {
            printf("Elemento [%d,%d]: ", i, j);
            scanf("%lf", &mat[i][j]);
        }
    }

    somaPositivos = 0;
    for (int i = 0; i < N; i++) {
        for (int j = 0; j < N; j++) {
            if (mat[i][j] > 0) {
                somaPositivos = somaPositivos + mat[i][j];
            }
        }
    }

    printf("\nSOMA DOS POSITIVOS: %.1lf\n\n", somaPositivos);

	printf("Escolha uma linha: ");
	scanf("%d", &indLinha);

	printf("LINHA ESCOLHIDA: ");

    for (int i = 0; i < N; i++) {
        printf("%.1lf ", mat[indLinha][i]);
    }

	printf("\n\nEscolha uma coluna: ");
	scanf("%d", &indColuna);

    printf("COLUNA ESCOLHIDA: ");

    for (int i = 0; i < N; i++) {
        printf("%.1lf ", mat[i][indColuna]);
    }

	printf("\n\nDIAGONAL PRINCIPAL: ");

    for (int i = 0; i < N; i++) {
        printf("%.1lf ", mat[i][i]);
    }

    for (int i = 0; i < N; i++) {
        for (int j = 0; j < N; j++) {
            if (mat[i][j] < 0) {
                mat[i][j] = pow(mat[i][j], 2);
            }
        }
    }

	printf("\n\nMATRIZ ALTERADA:\n");

    for (int i = 0; i < N; i++) {
        for (int j = 0; j < N; j++) {
            printf("%.1lf ", mat[i][j]);
        }
		printf("\n");
    }

    return 0;
}

File: python/matriz_geral.py
import math

N: int
indicaLinha: int
indicaColuna: int
soma: float

N = int(input("Qual a ordem da matriz? "))

mat: [[float]] = [[0 for x in range(N)] for x in range(N)]

for i in range(N):
    for j in range(N):
        mat[i][j] = float(input(f"Elemento [{i},{j}]: "))

soma = 0
for i in range(N):
    for j in range(N):
        if mat[i][j] > 0:
            soma += mat[i][j]

print()
print(f"SOMA DOS POSITIVOS: {soma:.1f}")
print()

indicaLinha = int(input("Escolha uma linha: "))

print("LINHA ESCOLHIDA: ", end="")

for i in range(N):
    print(f"{mat[indicaLinha][i]:.1f} ", end="")

print()
print()
indicaColuna = int(input("Escolha uma coluna: "))

print("COLUNA ESCOLHIDA: ", end="")

for i in range(N):
    print(f"{mat[i][indicaColuna]:.1f} ", end="")

print()
print()
print("DIAGONAL PRINCIPAL: ", end="")

for i in range(N):
    print(f"{mat[i][i]:.1f} ", end="")

for i in range(N):
    for j in range(N):
        if mat[i][j] < 0:
            mat[i][j] = math.pow(mat[i][j], 2)

print()
print()
print("MATRIZ ALTERADA:")

for i in range(N):
    for j in range(N):
        print(f"{mat[i][j]:.1f} ", end="")
    print()
